contracts/protocols/mainnet/augmented-finance/v1/helpers.sol

```diff
@@ -0,0 +1,216 @@
+     * @param dataProvider data provider contract `IProtocolDataProvider`
+     * @param user token holder
+     * @param token token address
+     * @param priceInETH token price in ETH
+     * @param priceInUSD token price is USD
+     */
+    function getTokenData(
+        IProtocolDataProvider dataProvider,
+        address user,
+        address token,
+        uint256 priceInETH,
+        uint256 priceInUSD
+    ) internal view returns (UserTokenData memory tokenData) {
+        TokenData memory data = _getTokenData(dataProvider, token);
+
+        (
+            tokenData.supplyBalance,
+            tokenData.stableBorrowBalance,
+            tokenData.variableBorrowBalance,
+            ,
+            ,
+            tokenData.userStableBorrowRate,
+            ,
+            ,
+            tokenData.isCollateral
+        ) = dataProvider.getUserReserveData(token, user);
+
+        (
+            data.availableLiquidity,
+            data.totalStableDebt,
+            data.totalVariableDebt,
+            tokenData.supplyRate,
+            tokenData.variableBorrowRate,
+            tokenData.stableBorrowRate,
+            ,
+            ,
+            ,
+
+        ) = dataProvider.getReserveData(token);
+
+        tokenData.priceInETH = priceInETH;
+        tokenData.priceInUSD = priceInUSD;
+        tokenData.data = data;
+    }
+
+    /**
+     * @dev Get extended token data from configurator
+     * @param dataProvider data provider contract `IProtocolDataProvider`
+     * @param token token address
+     */
+    function _getTokenData(IProtocolDataProvider dataProvider, address token)
+        private
+        view
+        returns (TokenData memory data)
+    {
+        (
+            ,
+            data.ltv,
+            data.threshold,
+            ,
+            data.reserveFactor,
+            data.usageAsCollateralEnabled,
+            data.borrowEnabled,
+            data.stableBorrrowEnabled,
+            data.isActive,
+            data.isFrozen
+        ) = dataProvider.getReserveConfigurationData(token);
+        (address depositTokenAddress, , ) = dataProvider.getReserveTokensAddresses(token);
+
+        data.totalSupply = IToken(depositTokenAddress).totalSupply();
+    }
+
+    /**
+     * @dev Get user data
+     * @param pool leding pool contract `ILendingPool`
+     * @param dataProvider data provider contract `IProtocolDataProvider`
+     * @param user user address
+     * @param ethPriceInUsd ETH price in USD
+     */
+    function getUserData(
+        ILendingPool pool,
+        IProtocolDataProvider dataProvider,
+        address user,
+        uint256 ethPriceInUsd
+    ) internal view returns (UserData memory userData) {
+        (
+            uint256 totalCollateralETH,
+            uint256 totalDebtETH,
+            uint256 availableBorrowsETH,
+            uint256 currentLiquidationThreshold,
+            uint256 ltv,
+            uint256 healthFactor
+        ) = pool.getUserAccountData(user);
```

Comment:
you can set struct fields directly, without intermediate variables as you do above